src/components/Analytics/AnalyticsMenu.js

```diff
@@ -0,0 +1,243 @@
+import { hooks } from '../../config/queryClient';
+import {
+  ANALYTICS_CONTAINER_CY,
+  ANALYTICS_GENERAL_TAB_MENU_CY,
+  buildAnalyticsDetailedQuestionTabMenuCy,
+} from '../../config/selectors';
+import { useElementWidth } from '../../hooks/useElementWidth';
+import {
+  useMaxAvailableHeightInWindow,
+  useMaxAvailableHeightWithParentHeight,
+} from '../../hooks/useMaxAvailableHeight';
+import { QuizContext } from '../context/QuizContext';
+import AutoScrollableMenu from '../navigation/AutoScrollableMenu';
+import TabPanel from '../navigation/TabPanel';
+import {
+  fillInTheBlankCharts,
+  generalCharts,
+  multipleChoicesCharts,
+  sliderCharts,
+  textInputCharts,
+} from './AnalyticsCharts';
+import QuestionDetailedCharts from './detailedCharts/QuestionDetailedCharts';
+import GeneralCharts from './genaralCharts/GeneralCharts';
+
+const SLIDE_BAR_WIDTH = 16;
+
+/**
+ * Component that represents the Analytics menu. Handle which charts have to be drawn.
+ *
+ * @param headerElem The header element if any, used to calculate the remaining height that this object can take
+ * in the window
+ */
+const AnalyticsMenu = ({ headerElem }) => {
+  const { t } = useTranslation();
+
+  // Fetch all the data here, as it is used by multiple children, avoid to fetch it in all children
+  const { questions, order } = useContext(QuizContext);
+  const { data, isLoading: isContextLoading } = hooks.useAppContext();
+  const { data: responses, isLoading } = hooks.useAppData();
+
+  // The charts to display, must contain 'link' and 'label' properties so that they can easily be passed
+  // to AutoScrollableMenu component
+  const [charts, setCharts] = useState(generalCharts(t));
+  const chartRefs = useRef({});
+  const chartContainerRef = useRef(null);
+  const [stackElem, setStackElem] = useState(null);
+  const [sideMenuElem, setSideMenuElem] = useState(null);
+  const chartTabs = useRef(null);
+  const maxResultViewHeight = useMaxAvailableHeightInWindow(headerElem.current);
+  const maxHeightScrollableMenu = useMaxAvailableHeightWithParentHeight(
+    maxResultViewHeight,
+    chartTabs.current
+  );
+  const [tab, setTab] = useState(0);
+
+  const questionById = useMemo(
+    () => questions.groupBy((q) => q.id),
+    [questions]
+  );
+
+  const stackElemWidth = useElementWidth(stackElem);
+  const sideMenuElemWidth = useElementWidth(sideMenuElem);
+
+  /**
+   * List of question ordered, so that we can easily get the question represented by the index of a tab
+   * useful to know the type of the question, to correctly get the charts to display for a question type
+   */
+  const detailedChartTabQuestion = useMemo(
+    () => order.map((qId) => questionById.get(qId)?.first()),
+    [order, questionById]
+  );
+
+  /**
+   * Callback to get the charts given the type of the question that has been selected
+   */
+  const getChartsForType = useCallback(
+    (question) => {
+      // eslint-disable-next-line default-case
```

Comment:
Maybe add a default case, but you `console.error`?

Reply:
Yes you are right, I added a default case that logs to the error console the received type if it didn't match any of the four possible types.